setup.py

```diff
@@ -64,230 +59,215 @@ def list_all_files_in_dir(path):
         if IS_WINDOWS:
-            return 'MinGW Makefiles'
+            return "MinGW Makefiles"
         return None
 
     def get_extra_libraries(self):
-        extra_libs = ['ssl']
+        extra_libs = ["ssl"]
         if IS_WINDOWS:
-            extra_libs.extend(['crypto', 'ws2_32'])
+            extra_libs.extend(["crypto", "ws2_32"])
         return extra_libs
 
-    def get_extra_cmake_config_args(self):
+    def get_openssl_root_dir(self):
         if not IS_APPLE:
-            return []
+            return None
 
-        openssl_root_dir_env_var = 'OPENSSL_ROOT_DIR'
+        openssl_root_dir_env_var = "OPENSSL_ROOT_DIR"
         openssl_root_dir = os.getenv(openssl_root_dir_env_var)
 
         if openssl_root_dir:
             self.announce(
-                f'Using the value of {openssl_root_dir_env_var} for OpenSSL,'
-                f'which is {openssl_root_dir}', level=log.INFO)
-            return [f'-DOPENSSL_ROOT_DIR={openssl_root_dir}']
+                f"Using the value of {openssl_root_dir_env_var} for OpenSSL," f"which is {openssl_root_dir}",
+                level=log.INFO,
+            )
+            return openssl_root_dir
+
+        def get_latest_openssl_subdirs(possible_root_dirs):
+            def get_latest_openssl_subdir(possible_root_dir):
+                if not os.path.isdir(possible_root_dir):
+                    # Maybe CMake can find OpenSSL properly without this
+                    return None
 
-        default_openssl_root_dir = '/usr/local/Cellar/openssl@1.1'
+                self.announce(f"Checking {possible_root_dir} for possible OpenSSL installation", level=log.INFO)
+                openssl_versions = sorted(pathlib.Path(possible_root_dir).iterdir(), key=os.path.getmtime, reverse=True)
 
-        if not os.path.isdir(default_openssl_root_dir):
-            # Maybe CMake can find OpenSSL properly without this
-            return []
+                if not openssl_versions:
+                    return None
 
-        self.announce('d', level=log.INFO)
-        openssl_versions = sorted(
-            pathlib.Path(default_openssl_root_dir).iterdir(),
-            key=os.path.getmtime,
-            reverse=True)
+                return str(openssl_versions[0])
 
-        if not openssl_versions:
-            return []
+            return [get_latest_openssl_subdir(p) for p in possible_root_dirs]
 
-        openssl_root_dir = str(openssl_versions[0])
-        self.announce(
-            f'Found OpenSSL in {openssl_root_dir}', level=log.INFO)
-        return [f'-DOPENSSL_ROOT_DIR={openssl_root_dir}']
+        possible_openssl_root_dirs = ["/opt/homebrew/Cellar/openssl@1.1", "/usr/local/Cellar/openssl@1.1"]
+
+        try:
+            openssl_root_dir = [p for p in get_latest_openssl_subdirs(possible_openssl_root_dirs) if p is not None][0]
+            self.announce(f"Found OpenSSL in {openssl_root_dir}", level=log.INFO)
+            return openssl_root_dir
+        except ValueError:
```

Comment:
If `openssl_root_dir` is an empty array (I think that could happen), `IndexError` (because of `[0]`) will be raised and not caught here?

Reply:
I think this should not be the case? macos comes with a preinstalled version of openssl ?

Reply:
macOS might come with libraries for openSSL, but not with the development files (headers, configs, etc), that why openSSL needs to be installed.

@gitbuda you are right, I will fix this.